OneSignalSDK/onesignal/src/main/java/com/onesignal/LocationController.java

```diff
@@ -248,40 +273,100 @@ else if (permissionList.contains("android.permission.ACCESS_COARSE_LOCATION")) {
 
    // Started from this class or PermissionActivity
    static void startGetLocation() {
-      // Prevents overlapping requests
-      if (fallbackFailThread != null)
-         return;
+      OneSignal.Log(OneSignal.LOG_LEVEL.DEBUG, "LocationController startGetLocation with lastLocation: " + lastLocation);
+
+      if (locationHandlerThread == null)
+         locationHandlerThread = new LocationHandlerThread();
 
       try {
-         synchronized (syncLock) {
-            startFallBackThread();
-
-            if (locationHandlerThread == null)
-               locationHandlerThread = new LocationHandlerThread();
-
-            if (mGoogleApiClient == null || mLastLocation == null) {
-               GoogleApiClientListener googleApiClientListener = new GoogleApiClientListener();
-               GoogleApiClient googleApiClient = new GoogleApiClient.Builder(classContext)
-                       .addApi(LocationServices.API)
-                       .addConnectionCallbacks(googleApiClientListener)
-                       .addOnConnectionFailedListener(googleApiClientListener)
-                       .setHandler(locationHandlerThread.mHandler)
-                       .build();
-               mGoogleApiClient = new GoogleApiClientCompatProxy(googleApiClient);
-
-               mGoogleApiClient.connect();
-            }
-            else if (mLastLocation != null)
-               fireCompleteForLocation(mLastLocation);
+         if (isGooglePlayServicesAvailable(classContext)) {
```

Comment:
We will want to change this to the same logic we are using the `getDeviceType` once PR #1026 is merged in.

Reply:
Rebased this PR off of `huawei_handle_notif_opens` so we can address this now.
@Jeasmine I addressed this in PR #1045 if you can do a review to close the comment